src/libs/Navigation/Navigation.js

```diff
@@ -125,13 +125,21 @@ function dismissModal(shouldOpenDrawer = false) {
 /**
  * Check whether the passed route is currently Active or not.
  *
+ * Building path with useLinkBuilder since navigationRef.current.getCurrentRoute().path
+ * is undefined in the first navigation.
+ *
  * @param {String} routePath Path to check
  * @return {Boolean} is active
  */
-function isActive(routePath) {
+function isActiveRoute(routePath) {
+    const buildLink = useLinkBuilder();
+
     // We remove First forward slash from the URL before matching
-    const path = navigationRef.current && navigationRef.current.getCurrentRoute().path
-        ? navigationRef.current.getCurrentRoute().path.substring(1)
+    const path = navigationRef.current && navigationRef.current.getCurrentRoute().name
+        ? buildLink(
+            navigationRef.current.getCurrentRoute().name,
+            navigationRef.current.getCurrentRoute().params,
+        ).substring(1)
```

Comment:
We should have kept using `getCurrentRoute().path` as the first resource and only fallback to `buildLink` if needed. This is because the `useLinkBuilder` or `getPathFromState` does not work well with urls that contain encoded parameters. `react-navigation` encodes every param even the encoded ones resulting in double-encoded urls. Those double-encoded urls won't match with the user's urls. - Coming from https://github.com/Expensify/App/issues/33001